pricenode/src/main/java/bisq/price/spot/ExchangeRateService.java

```diff
@@ -53,26 +60,88 @@ public ExchangeRateService(List<ExchangeRateProvider> providers) {
+            // Therefore, add metadata fields for all known providers
+            // Rates are encapsulated in the "data" map below
             metadata.putAll(getMetadata(p, exchangeRates));
-            exchangeRates.forEach(e ->
-                allExchangeRates.put(e.getCurrency(), e)
-            );
         });
 
         return new LinkedHashMap<String, Object>() {{
             putAll(metadata);
             // Use a sorted list by currency code to make comparision of json data between different
             // price nodes easier
-            List<ExchangeRate> values = new ArrayList<>(allExchangeRates.values());
+            List<ExchangeRate> values = new ArrayList<>(aggregateExchangeRates.values());
             values.sort(Comparator.comparing(ExchangeRate::getCurrency));
             put("data", values);
         }};
     }
 
+    /**
+     * For each currency, create an aggregate {@link ExchangeRate} based on the currency's rates from all providers.
+     * If multiple providers have rates for the currency, then aggregate price = average of retrieved prices.
+     * If a single provider has rates for the currency, then aggregate price = the rate from that provider.
+     *
+     * @return Aggregate {@link ExchangeRate}s based on info from all providers, indexed by currency code
+     */
+    private Map<String, ExchangeRate> getAggregateExchangeRates() {
+        Map<String, ExchangeRate> aggregateExchangeRates = new HashMap<>();
+
+        // Query all known providers and collect all exchange rates, grouped by currency code
+        Map<String, List<ExchangeRate>> currencyCodeToExchangeRates = getCurrencyCodeToExchangeRates();
+
+        // For each currency code, calculate aggregate rate
+        currencyCodeToExchangeRates.forEach((currencyCode, exchangeRateList) -> {
+            ExchangeRate aggregateExchangeRate;
+            if (exchangeRateList.size() == 1) {
+                // If a single provider has rates for this currency, then aggregate = rate from that provider
+                aggregateExchangeRate = exchangeRateList.get(0);
+            }
+            else if (exchangeRateList.size() > 1) {
+                // If multiple providers have rates for this currency, then aggregate = average of the rates
+                OptionalDouble opt = exchangeRateList.stream().mapToDouble(ExchangeRate::getPrice).average();
+                double priceAvg = opt.orElseThrow(IllegalStateException::new); // List size > 1, so opt is always set
+
+                aggregateExchangeRate = new ExchangeRate(
+                        currencyCode,
+                        BigDecimal.valueOf(priceAvg),
+                        new Date(), // timestamp = time when avg is calculated
+                        "Bisq-Aggregate");
+            }
+            else {
+                // If the map was built incorrectly and this currency points to an empty list of rates, skip it
+               return;
+            }
```

Comment:
Prefer modelling this as guard logic, e.g.:

```java
if (exchangeRateList.isEmpty())
    // If the map was built incorrectly and this currency points to an empty list of rates, skip it
    return;

if (exchangeRateList.size() == 1) {
    // If a single provider has rates for this currency, then aggregate = rate from that provider
    aggregateExchangeRate = exchangeRateList.get(0);
}
else {
    // If multiple providers have rates for this currency, then aggregate = average of the rates
    // ...
}
aggregateExchangeRates.put(aggregateExchangeRate.getCurrency(), aggregateExchangeRate);
```

See also https://github.com/bisq-network/style/issues/12